src/ObjectUtils/PdbFileLlvm.cpp

```diff
@@ -214,6 +257,121 @@ PdbFileLlvm::PdbFileLlvm(std::filesystem::path file_path, const ObjectFileInfo&
                           llvm::toString(std::move(error)))};
     }
   }
+  return outcome::success();
+}
+
+void LoadDebugSymbolsFromPublicSymbolStream(
+    llvm::pdb::PublicsStream& public_symbol_stream, llvm::pdb::SymbolStream& symbol_stream,
+    llvm::FixedStreamArray<llvm::object::coff_section>& section_headers,
+    const ObjectFileInfo& object_file_info, std::vector<SymbolInfo>* symbol_infos,
+    absl::flat_hash_set<uint64_t>* addresses_from_module_debug_stream) {
+  const llvm::pdb::GSIHashTable& public_symbol_has_records = public_symbol_stream.getPublicsTable();
+  for (const auto& hash_record : public_symbol_has_records) {
+    llvm::Expected<llvm::codeview::PublicSym32> record =
+        llvm::codeview::SymbolDeserializer::deserializeAs<llvm::codeview::PublicSym32>(
+            symbol_stream.readRecord(hash_record));
+    ORBIT_CHECK(record);
+
+    // Skip this symbol if it is not a function (but rather a global constant).
+    if ((record->Flags & llvm::codeview::PublicSymFlags::Function) ==
+        llvm::codeview::PublicSymFlags::None) {
+      continue;
+    }
+
+    SymbolInfo symbol_info;
+    uint64_t address = ComputeAddress(record->Offset, record->Segment, object_file_info.load_bias,
+                                      section_headers);
+    symbol_info.set_address(address);
```

Comment:
I would put this below `set_demangled_name` and do `addresses_from_module_debug_stream->contains` before creating a `SymbolInfo`.

Reply:
Sounds better, thanks.